crates/cairo-addons/src/vm/vm_consts.rs

```diff
@@ -322,15 +320,16 @@ impl PyVmConst {
         match &self.var.var_type {
             CairoVarType::Pointer { .. } => {
                 // The address of the pointer is the same as its value.
-                let addr = self
-                    .var
-                    .value
-                    .as_ref()
-                    .ok_or_else(|| PyAttributeError::new_err("Pointer has no address"))?
-                    .get_relocatable()
-                    .expect("Pointer value is not relocatable");
-
-                Ok(Some(addr))
+                // Note: if the variable is NOT a pointer (e.g. we cast something to a pointer),
+                // like tempvar my_pointer_struct = MyPointerStruct(cast(0, felt*));
+                // then the value is 0 and we should return `self.var.address`
+                let pointer_value = self.var.value.as_ref();
+
+                match pointer_value {
+                    Some(MaybeRelocatable::RelocatableValue(rel)) => Ok(Some(*rel)),
+                    Some(MaybeRelocatable::Int(_)) => Ok(self.var.address),
```

Comment:
are we saying that here, we expect `Some(MaybeRelocatable::Int(0))`?

should we use only Some(MaybeRelocatable::Int(0)) and fail otherwise?

Reply:
Technically it could be any non pointer value not just 0

Reply:
it's not based on what "we" want it's based on VM rules, you can cast any felt to a felt*